gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -496,6 +499,20 @@ private ManagedChannel createSingleChannel() throws IOException {
     return managedChannel;
   }
 
+  // dedup any headers explicitly set with headers provided by credential, with preference to
+  // credential headers
+  private Map<String, String> mergeHeadersWithCredentialHeaders() {
+    Map<String, String> userHeaders = new HashMap<>(headerProvider.getHeaders());
+    if (credentials != null) {
+      try {
+        userHeaders.keySet().removeAll(credentials.getRequestMetadata().keySet());
+      } catch (Exception e) {
```

Comment:
you're correct, updated it to catch IO, but I left as a try catch because I didn't think we wanted to fail the whole call in case of an error requesting the credentials here.  In that case if they couldn't request them when actually making the call it would still fail, but otherwise if they can get retrieve it during the actual grpc call, it would succeed.